### PR TITLE
Add decaf448

### PR DESCRIPTION
**Pull request by stknob**
Based on draft-irtf-cfrg-ristretto255-decaf448-07, draft-irtf-cfrg-hash-to-curve-16 and the ristretto255 implementation.

### CODE REVIEW

**Comment by sublimator** [comment]
nice :)

File: test/ed448-addons.test.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      // These are all bad because they give a nonsquare x².
+      '58ad48715c9a102569b68b88362a4b0645781f5a19eb7e59c6a4686fd0f0750ff42e3d7af1ab38c29d69b670f31258919c9fdbf6093d06c0',
+      '8ca37ee2b15693f06e910cf43c4e32f1d5551dda8b1e48cb6ddd55e440dbc7b296b601919a4e4069f59239ca247ff693f7daa42f086122b1',
+      '982c0ec7f43d9f97c0a74b36db0abd9ca6bfb98123a90782787242c8a523cdc76df14a910d54471127e7662a1059201f902940cd39d57af5',
+      'baa9ab82d07ca282b968a911a6c3728d74bf2fe258901925787f03ee4be7e3cb6684fd1bcfe5071a9a974ad249a4aaa8ca81264216c68574',
+      '2ed9ffe2ded67a372b181ac524996402c42970629db03f5e8636cbaf6074b523d154a7a8c4472c4c353ab88cd6fec7da7780834cc5bd5242',
+      'f063769e4241e76d815800e4933a3a144327a30ec40758ad3723a788388399f7b3f5d45b6351eb8eddefda7d5bff4ee920d338a8b89d8b63',
+      '5a0104f1f55d152ceb68bc138182499891d90ee8f09b40038ccc1e07cb621fd462f781d045732a4f0bda73f0b2acf94355424ff0388d4b9c',
+    ];
+    for (const badBytes of badEncodings) {
```
Comment (sublimator) [inline test/ed448-addons.test.js:68]: there's nothing like a should.each ?

File: test/ed448-addons.test.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      '20b171cb16be977f15e013b9752cf86c54c631c4fc8cbf7c03c4d3ac9b8e8640e7b0e9300b987fe0ab5044669314f6ed1650ae037db853f1',
+    ];
+
+    for (let i = 0; i < hashes.length; i++) {
+      const hash = hexToBytes(hashes[i]);
+      const point = DecafPoint.hashToCurve(hash);
+      deepStrictEqual(point.toHex(), encodedHashToPoints[i]);
+    }
+  });
+  should('have proper equality testing', () => {
```
Comment (sublimator) [inline test/ed448-addons.test.js:99]: This is just for testing DecafPoint#equals ?

**Comment by paulmillr** [comment]
This is great!

**Comment by paulmillr** [comment]
Tests seem to be failing.

File: test/ed448-addons.test.js (hunk 1 above)
Comment (paulmillr) [reply test/ed448-addons.test.js:68]: nah, no need to complicate it

File: test/ed448-addons.test.js (hunk 2 above)
Comment (stknob) [reply test/ed448-addons.test.js:96]: Jup, basic testing of the equals method (based on the RistrettoPoint one).

**Comment by stknob** [comment]
> Tests seem to be failing.

Forgot to run prettier on the files, should be fixed now.

File: test/ed448-addons.test.js (hunk 1 above)
Comment (sublimator) [reply test/ed448-addons.test.js:65]: I guess if you really wanted to, you can just do "arr.forEach((e, i) => should(`something - ${i} - ${e}`, " but then again "throws(() => something(b), badBytes)" will report the exact culprit. So yeah

**Comment by paulmillr** [comment]
Add benchmarks please and we're good to go.

**Comment by paulmillr** [comment]
Also, README docs.

**Comment by paulmillr** [comment]
Thanks!

In the future, please don't force-push: this requires me to re-review all files for the second time, instead of just watching new/diff commits.

If you wanted to ensure history cleanliness, it's good, but no need: I can always squash-merge a PR if it gets too messy.

**Comment by sublimator** [comment]
Nice! can use this for voprf

**Pull request merged**